core/src/main/java/bio/terra/pearl/core/service/export/EnrolleeImportService.java

```diff
@@ -524,14 +525,26 @@ protected SurveyResponse importPreEnrollResponse(Survey preEnroll,
         return created;
     }
 
-    protected SurveyResponse importSurveyResponse(UUID portalId, SurveyFormatter formatter, Map<String, String> enrolleeMap, ExportOptions exportOptions,
+    protected List<SurveyResponse> importSurveyResponses(UUID portalId, SurveyFormatter formatter, Map<String, String> enrolleeMap, ExportOptions exportOptions,
                                                   StudyEnvironment studyEnv, PortalParticipantUser ppUser, Enrollee enrollee, DataAuditInfo auditInfo) {
-        SurveyResponse response = formatter.fromStringMap(studyEnv.getId(), enrolleeMap);
-        if (response == null) {
+        List<SurveyResponseWithTaskDto> responses = formatter.listFromStringMap(studyEnv.getId(), enrolleeMap);
+        if (responses == null || responses.isEmpty()) {
             return null;
         }
-        ParticipantTask relatedTask = participantTaskService.findTaskForActivity(ppUser.getId(), studyEnv.getId(), formatter.getModuleName())
-                .orElse(null);
+
+        List<SurveyResponse> imported = new ArrayList<>();
+        for (int i = 0; i < responses.size(); i++) {
+            SurveyResponseWithTaskDto response = responses.get(i);
+            imported.add(importSurveyResponse(ppUser, enrollee, studyEnv, formatter, response, portalId, auditInfo, enrolleeMap, i + 1));
+        }
+
+        return imported;
+    }
+
+    private SurveyResponse importSurveyResponse(PortalParticipantUser ppUser, Enrollee enrollee, StudyEnvironment studyEnv, SurveyFormatter formatter, SurveyResponseWithTaskDto response, UUID portalId, DataAuditInfo auditInfo, Map<String, String> enrolleeMap, Integer repeatNum) {
+        // if its the first repeat, attempt to find existing task to update. other repeats are previous responses and need their own task
+        ParticipantTask relatedTask = repeatNum == 1 ? participantTaskService.findTaskForActivity(ppUser.getId(), studyEnv.getId(), formatter.getModuleName())
```

Comment:
I am not sure if this is the correct, desired logic.

Currently, survey response import can also function to update existing responses, i.e. it is idempotent. However, if we import response history, this gets tricky - there would be multiple tasks in the database with no clear way to get the one you want to update.

So, this ternary right here makes survey response no longer idempotent - if you import the same participant multiple times, they will get an increasingly large number of new surveyresponse/tasks.

Reply:
Hmm... making this non-idempotent means fixing up an existing import will be a lot harder.  Can the logic be something like:
```
if (repeatNum == 1 && the response doesn't have an explicit completed at date ) {
 auto-match with findTaskForActivity
} else if (response has an explicit completedAt date) {
 match task based on stableId and completedAt
} else {
  find all the tasks for that stableId sorted by date, and take the repeatNumTh task.  If there isn't one, create a new one.
}